app/lib/features/attachments/widgets/attachment_item.dart

```diff
@@ -1,41 +1,188 @@
+        border: Border.all(color: borderColor),
+      ),
+      child: ListTile(
+        leading:
+            mediaState.mediaLoadingState.isLoading || mediaState.isDownloading
+                ? loadingIndication()
+                : attachmentLeadingIcon(attachmentType),
+        onTap: () => attachmentOnTap(
+          ref,
+          context,
+          attachmentType,
+          mediaState,
+        ),
+        onLongPress: canEdit
+            ? () => openRedactContentDialog(
+                  context,
+                  eventId: eventId,
+                  roomId: roomId,
+                  title: L10n.of(context).deleteAttachment,
+                  description: L10n.of(context)
+                      .areYouSureYouWantToRemoveAttachmentFromPin,
+                  isSpace: true,
+                )
+            : null,
+        title: title(attachmentType),
+        trailing: Visibility(
+          visible: mediaState.mediaFile == null &&
+              attachmentType != AttachmentType.link,
+          child: IconButton(
+            onPressed: () => attachmentOnTap(
+              ref,
+              context,
+              attachmentType,
+              mediaState,
+            ),
+            icon: const Icon(Atlas.download_arrow_down),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget title(AttachmentType attachmentType) {
+    final msgContent = attachment.msgContent();
+    final fileName = msgContent.body();
+    final fileNameSplit = fileName.split('.');
+    final title = fileNameSplit.first;
+    final fileExtension = fileNameSplit.last;
+    String fileSize = getFileSize(msgContent.size() ?? 0);
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        if (attachmentType == AttachmentType.link) ...[
+          Text(
+            attachment.name() ?? '',
+            maxLines: 1,
+            overflow: TextOverflow.ellipsis,
+          ),
+          Text(
+            attachment.link() ?? '',
+            maxLines: 2,
+            overflow: TextOverflow.ellipsis,
+          ),
+        ] else ...[
+          Text(
+            title,
+            maxLines: 2,
+            overflow: TextOverflow.ellipsis,
+          ),
+          Row(
+            children: [
+              Text(fileSize),
+              const SizedBox(width: 10),
+              const Text('.'),
+              const SizedBox(width: 10),
+              Text(documentTypeFromFileExtension(fileExtension)),
+            ],
+          ),
+        ],
+      ],
+    );
+  }
+
+  Future<void> attachmentOnTap(
+    WidgetRef ref,
+    BuildContext context,
+    AttachmentType attachmentType,
+    AttachmentMediaState mediaState,
+  ) async {
+    // Open attachment link
+    if (attachmentType == AttachmentType.link) {
+      openLink(attachment.link() ?? '', context);
+    }
+
+    // If attachment is media then check media is downloaded
+    else if (mediaState.mediaFile == null) {
```

Comment:
let's keep if and else together pls
```suggestion
    } else if (mediaState.mediaFile == null) {
    // If attachment is media then check media is downloaded
```